### PR TITLE
[FIX] use domain when loading new partner in Point of Sale

### PR DESCRIPTION
**Pull request by legalsylvain**
Description of the issue/feature this PR addresses:

the domain for the res.partner model is used when PoS is loaded. but an other call is done to load new partners, since the last load of the PoS. this call doesn't reuse domain.

Current behavior before PR:

All partners are loaded.

Desired behavior after PR is merged:

Only desired partners are loaded.

CC : 
- @NL66278 (for  pos_restricted_customer_list) and #168 reviewiers (@rafaelbn, @hparfr) 
- @StefanRijnhart 

regards.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -443,9 +443,11 @@ function openerp_pos_models(instance, module){ //module is instance.point_of_sal
             var self = this;
             var def  = new $.Deferred();
             var fields = _.find(this.models,function(model){ return model.model === 'res.partner'; }).fields;
+            var domain = _.find(this.models, function(model){ return model.model === 'res.partner'; }).domain.slice(0);
+            domain.push(['write_date','>',this.db.get_partner_write_date()]);
```
Comment (hparfr) [inline addons/point_of_sale/static/src/js/models.js:447]: why slice(0) ?

Comment (legalsylvain) [reply addons/point_of_sale/static/src/js/models.js:447]: to make a copy of the list. This function will be called many times, if slice(0) is not set, 
- the original domain will be changed (not desired),
- the domain will become longer each time.

Ref : https://stackoverflow.com/questions/15722433/javascript-copy-array-to-new-array

regards.

**Review by hparfr — APPROVED** [review]

**Comment by rafaelbn** [comment]
Hi @legalsylvain , I cannot help here too much. This PR need technical review

Note that travis is :red_circle:

**Comment by legalsylvain** [comment]
Hi @rafaelbn. If you want, you can test it. 

- load the PoS 
- create a supplier (non customer) 
- display customers in the PoS 

before : 
the supplier is displayed. (undesired behaviour)

after
the supplier is not loaded.

Note : Travis is red for unrelated reasons.

regards.

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1: 

@pedrobaeza could we merge this?

**Comment by pedrobaeza** [comment]
At least this should be rebased to have a green Travis.

**Comment by StefanRijnhart** [comment]
Does this apply to later versions of Odoo, or has it been fixed there? If the issue still occurs, we should propose a similar fix to that later version.

**Comment by pedrobaeza** [comment]
Please rebase to get a green build

**Comment by legalsylvain** [comment]
@pedrobaeza : rebase done.

@StefanRijnhart : I only use 8.0 for instance, so I didn't have feedback on most recent version. But reading the code of the master branch : 
https://github.com/odoo/odoo/blob/master/addons/point_of_sale/static/src/js/models.js#L596

- the domain now include ```is_customer == True``` --> good news ! (consistent with the initial load)
- the domain is still static. (doesn't reuse domain defined during the initial load (*) and that can be changed by custom module(**)) --> still bad, and should be fixed, but propose a PR on odoo will maybe not be accepted, because it's a problem only for developpers that realize custom modules. (not an "Odoo bug" now).

(*) : domain originally defined here : https://github.com/odoo/odoo/blob/master/addons/point_of_sale/static/src/js/models.js#L190

(**) : a sample of a modification of the res.partner domain : https://github.com/grap/grap-odoo-business/blob/8.0/recurring_consignment/static/src/js/recurring_consignment.js#L20

**Comment by pedrobaeza** [comment]
Travis is still red after restarting it several times, so it seems there's something wrong in this PR.

**Comment by legalsylvain** [comment]
@pedrobaeza : 

Sorry but the message really seems unrelated to this PR, doesnt it ?

https://travis-ci.org/OCA/OCB/jobs/388184840#L1829

and current 8.0 branch is broken with similar message.

https://travis-ci.org/OCA/OCB/branches

**Comment by pedrobaeza** [comment]
It's not the same problem, as it's in another build. This kind of errors happen usually when there's an error on one test and the cursor get unusable, blocking the rest of operations. The problem is that Odoo doesn't inform about the problem. Have you launched the point_of_sale tests in local?

**Comment by legalsylvain** [comment]
Hi @pedrobaeza. I tested locally and all the tests are passing.

image

In the other hand, the test of the OCA 8.0 branch is [broken](https://travis-ci.org/OCA/OCB/jobs/403760744#L1810), with exactly the same message as the message of the [current Pull request](https://travis-ci.org/OCA/OCB/jobs/388184840#L1810). 

So, I can not figure out how this PR can break something.
Furthermore, I do not see either how changing 4 lines in PoS js file can set the cursor unusable.

I propose to merge this trivial fix that is working fine in production since 6 monthes for my side. 

thanks for your answer.

**Comment by StefanRijnhart** [comment]
Agree with @legalsylvain. Same message, same breakage, and only the localization job fails.

**Review by StefanRijnhart — APPROVED** [review]

**Comment by legalsylvain** [comment]
thanks !

**Pull request merged**